### ISSUE TITLE
VPC network not removed with `terraform destroy`

### ISSUE THREAD

**Issue by jackwootton** [terraform-google-modules/terraform-google-network#352]
### TL;DR

Creating a VPC Network using the `network` module also creates a VPC network **peer**. However, when running `terraform destroy` the network peer is not destroyed, therefore the VPC network itself is not destroyed because the _network peer_ depends on the _network_.

### Expected behavior

Running `terraform destroy` removes the VPC network.

### Observed behavior

The VPC network is not removed.

### Terraform Configuration

```hcl
# Install the required providers.
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.11.0"
    }
  }
}

# Data source to access the configuration of the Google Cloud provider.
data "google_client_config" "default" {}

# Create the VPC Network and subnetworks.
module "network" {
  source       = "registry.terraform.io/terraform-google-modules/network/google"
  version      = "5.0.0"
  network_name = "vpc"
  project_id   = var.project_id
  routing_mode = "GLOBAL"
  subnets = [
    {
      subnet_name           = var.vpc_kubernetes_subnet_name
      subnet_ip             = "10.10.0.0/16"
      subnet_region         = var.region
      subnet_private_access = "true"
    },
    {
      subnet_name           = var.vpc_sql_subnet_name
      subnet_ip             = "10.40.0.0/16"
      subnet_region         = var.region
      subnet_private_access = "true"
    }
  ]
  secondary_ranges = {
    (var.vpc_kubernetes_subnet_name) = [
      {
        range_name    = var.pods_ip_range_name
        ip_cidr_range = "10.20.0.0/16"
      },
      {
        range_name    = var.services_ip_range_name
        ip_cidr_range = "10.30.0.0/16"
      }
    ]
  }
}
```

### Terraform Version
```sh
Terraform v1.1.6
on linux_386
+ provider registry.terraform.io/hashicorp/external v2.2.0
+ provider registry.terraform.io/hashicorp/google v4.11.0
+ provider registry.terraform.io/hashicorp/google-beta v4.12.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.8.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/time v0.7.2

Your version of Terraform is out of date! The latest version
is 1.1.7. You can update by downloading from https://www.terraform.io/downloads.html
```

### Additional information

Details of the VPC peer network that's created
- `Name:`  servicenetworking-googleapis-com
- `VPC network:` vpc
- `Peered VPC network:` servicenetworking
